solver/llbsolver/solver.go

```diff
@@ -253,31 +253,42 @@ func (s *Solver) Solve(ctx context.Context, id string, sessionID string, req fro
 					}
 				}
 				for k, res := range crMap {
-					dtic, err := inlineCache(ctx, exp.CacheExporter, res, e.Config().Compression, session.NewGroup(sessionID))
+					dtic, err := inlineCache(ctx, out.CacheExporter, res, exp.Config().Compression, session.NewGroup(sessionID))
 					if err != nil {
 						return err
 					}
 					if dtic != nil {
 						inp.Metadata[fmt.Sprintf("%s/%s", exptypes.ExporterInlineCache, k)] = dtic
 					}
 				}
-				exp.CacheExporter = nil
+				out.CacheExporter = nil
 				return nil
 			}); err != nil {
 				return nil, err
 			}
 		}
-		if err := inBuilderContext(ctx, j, e.Name(), "", func(ctx context.Context, _ session.Group) error {
-			exporterResponse, err = e.Export(ctx, inp, j.SessionID)
-			return err
+		if err := inBuilderContext(ctx, j, exp.Name(), "", func(ctx context.Context, _ session.Group) error {
+			resp, err := exp.Export(ctx, inp, j.SessionID)
```

Comment:
I wonder if these should run in parallel if multiple set.

Reply:
I will give it a try and see if it can be made to run in parallel. Hopefully there's no mutable state to share.